prusti-encoder/src/util.rs

```diff
@@ -0,0 +1,124 @@
+use prusti_rustc_interface::{
```

Comment:
Please rename this file. It is very tempting to start putting random miscellaneous functions into a `util.rs` file when one exists. For now everything here seems to be related to types at least. Could it go into the `type` module? Could this be called `most_generic_ty.rs`?